crates/ty_python_semantic/src/types/instance.rs

```diff
@@ -155,21 +156,33 @@ impl<'db> From<NominalInstanceType<'db>> for Type<'db> {
 /// A `ProtocolInstanceType` represents the set of all possible runtime objects
 /// that conform to the interface described by a certain protocol.
 #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, PartialOrd, Ord, salsa::Update)]
-pub struct ProtocolInstanceType<'db>(
+pub struct ProtocolInstanceType<'db> {
+    pub(super) inner: Protocol<'db>,
+
     // Keep the inner field here private,
     // so that the only way of constructing `ProtocolInstanceType` instances
     // is through the `Type::instance` constructor function.
-    Protocol<'db>,
-);
+    _phantom: PhantomData<()>,
+}
 
 impl<'db> ProtocolInstanceType<'db> {
-    pub(super) fn inner(self) -> Protocol<'db> {
-        self.0
+    fn from_class(class: ClassType<'db>) -> Self {
```

Comment:
```suggestion
    // Keep this field private, so that the only way of constructing `ProtocolInstanceType` instances
    // is through the `Type::instance` constructor function.
    fn from_class(class: ClassType<'db>) -> Self {
```